.devcontainer/Dockerfile

```diff
@@ -0,0 +1,17 @@
+#syntax=docker/dockerfile:1.6
+
+FROM mcr.microsoft.com/vscode/devcontainers/python:3.12
+
+RUN <<EOF
+set -x
+set -e
+apt-get update
+apt-get install -yq --no-install-recommends \
+  ffmpeg \
+  ;
+apt-get clean
+rm -rf /var/lib/apt/lists/*
+pip install -U \
+  pre-commit \
+  ;
```

Comment:
pre-commit is installed with the test requirements

Reply:
Doesn't it need to be installed globally, rather than in the venv? That was the point of this implementation. If not, I'll commit your suggestions

Reply:
nope, in the venv.

Reply:
Done